lib/backend/model/bgppeer.go

```diff
@@ -31,11 +31,13 @@ var (
 	matchGlobalBGPPeer = regexp.MustCompile("^/?calico/bgp/v1/global/peer_v./([^/]+)$")
 	matchHostBGPPeer   = regexp.MustCompile("^/?calico/bgp/v1/host/([^/]+)/peer_v./([^/]+)$")
 	typeBGPPeer        = reflect.TypeOf(BGPPeer{})
+	ipPortSeparator    = "-"
 )
 
 type NodeBGPPeerKey struct {
 	Nodename string `json:"-" validate:"omitempty"`
 	PeerIP   net.IP `json:"-" validate:"required"`
+	Port     string `json:"-" validate:"omitempty"`
```

Comment:
I don't think we should use string for the type of Port. I'd be inclined to use `uint16` since it naturally validates.

Reply:
The reason for using string here was, we get port in `<ip>:<port>` format https://github.com/projectcalico/libcalico-go/blob/master/lib/apis/v3/bgppeer.go#L51 and later use net.SplitHostPort to split them, which returns port as string. Port used here is just to generate key, to avoid converting between string to uint16 and the back to string, I went for this. But looks like we do the same for IP, I'll update it to uint16

Reply:
The key is part of the public API so it's worth the extra conversion work to make the definition of key more "correct"